lib/logmpx.c

```diff
@@ -89,6 +89,7 @@ log_multiplexer_queue(LogPipe *s, LogMessage *msg, const LogPathOptions *path_op
   if (_has_multiple_arcs(self))
     {
       log_msg_write_protect(msg);
+      filterx_scope_write_protect(path_options->filterx_scope);
```

Comment:
I think this causes the crash. We might not have a `filterx_scope` here.

Reply:
Yup, thanks for catching this. I fixed this in my updated version.